tests/projection_test.py

```diff
@@ -45,6 +49,14 @@ def test_projection_box(self):
     self.assertArraysEqual(projection.projection_box((x, x), params),
                            (expected, expected))
 
+    # lower and upper are pytrees. Note: Does not work when U and L are scalars!
```

Comment:
I think the following could potentially work:
```python
if isinstance(lower_bound, float):
  # If lower_bound is a scalar, create a pytree with the same structure as `x`
  lower_bound = tree_util.tree_zeros_like(x) + lower_bound
```
but let's do it in a separate PR.